galaxy_ng/tests/integration/api/test_rbac_roles.py

```diff
@@ -0,0 +1,316 @@
+        index_exec_env,
+    },
+    "galaxy.execution_environment_publisher": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        change_exec_env_desc,
+        change_exec_env_readme,
+        # create_containers_under_existing_container_namespace,
+        # push_containers_to_existing_container_namespace,
+        # change_container_namespace,
+        # tag_untag_container_namespace,
+        # sync_remote_container,
+        create_exec_env,
+    },
+    "galaxy.execution_environment_namespace_owner": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        # change_exec_env_desc_object,
+        # change_exec_env_readme_object,
+        # create_containers_under_existing_container_namespace_object,
+        # push_containers_to_existing_container_namespace_object,
+        # change_container_namespace_object,
+        # tag_untag_container_namespace_object,
+        # sync_remote_container_object,
+    },
+    "galaxy.execution_environment_collaborator": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        # change_exec_env_desc_object,
+        # change_exec_env_readme_object,
+        # push_containers_to_existing_container_namespace_object,
+        # tag_untag_container_namespace_object,
+        # sync_remote_container_object,
+    },
+    "galaxy.group_admin": {
+        upload_collection_to_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        add_groups,
+        change_groups,
+        delete_groups,
+    },
+    "galaxy.user_admin": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_tasks,  # error?
+        view_groups,
+        view_users,
+        delete_users,
+        add_users,
+        change_users,
+    },
+    "galaxy.task_admin": {
+        upload_collection_to_namespace,  # error?
+        create_collection_namespace,  # error?
+        view_sync_configuration,  # error?
+        view_groups,
+        view_tasks,
+    }
+}
+
+
+@pytest.mark.role_rbac
+@pytest.mark.parametrize("role", ROLES_TO_TEST)
+def test_role_actions(role):
+    USERNAME = f"{NAMESPACE}_user_{gen_string()}"
+
+    user = create_user(USERNAME, PASSWORD)
+    group_id = create_group_with_user_and_role(user, role)
+
+    expected_allows = ROLES_TO_TEST[role]
+
+    # Test global actions
+    for action in GLOBAL_ACTIONS:
+        expect_pass = action in expected_allows
+        action(user, PASSWORD, expect_pass)
```

Comment:
Something like this would let you easily see which actions failed:

```suggestion
    failures = []

    # Test global actions
    for action in GLOBAL_ACTIONS:
        expect_pass = action in expected_allows
        try:
            action(user, PASSWORD, expect_pass)
        except AssertionError:
            failures.append(action.__name__)

    assert failures == []
```

This outputs the following when the test fails, which is a little more useful

```
>       assert failures == []
E       AssertionError: assert ['delete_coll..._collections'] == []
E         Left contains 5 more items, first extra item: 'delete_collection'
E         Full diff:
E           [
E         -  ,
E         +  'delete_collection',
E         +  'approve_collections',
E         +  'reject_collections',...
E
E         ...Full output truncated (4 lines hidden), use '-vv' to show
```